File: nysol_python/mcmd/methods/convTex/db/smp/mnewrand.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mnewrand'

############################### IDATA
db['idatas']=[]

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
実数乱数を10行生成する。乱数の種は1に固定しているので、いつ実行しても乱数系列は同じになる。
'''
script['sh_code']='''
mnewrand a=rand S=1 o=rsl1.csv
'''
script['py_code']='''
nm.mnewrand(a="rand", S="1", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='整数乱数'
script['comment']='''
最小値が0、最大値が1000、乱数の種が1の整数乱数を5行作成する。
'''
script['sh_code']='''
mnewrand a=rand -int max=1000 min=0 l=5 S=1 o=rsl2.csv
'''
script['py_code']='''
nm.mnewrand(a="rand", int=True, max="1000", min="0", l="5", S="1", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='ヘッダ行なしで出力'
script['comment']='''
``nfn=True`` でヘッダーなしのデータが生成される。
'''
script['sh_code']='''
mnewrand -nfn l=5 S=1 o=rsl3.csv
'''
script['py_code']='''
nm.mnewrand(nfn=True, l="5", S="1", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mnewrand.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

############## Example 1
title="Basic Example"
comment=<<'EOF'
Generate 10 rows of random integers. Use a fixed random seed so that it will always return the same sequence of random numbers.
EOF
scp=<<'EOF'
mnewrand a=rand S=1 o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Random Integers"
comment=<<'EOF'
Use random seed 1 to generate 5 rows of random integers with minimum value of 10 and maximum value of 100.
EOF
scp=<<'EOF'
mnewrand a=rand -int max=1000 min=0 l=5 S=1 o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Generate Output without Header"
comment=<<'EOF'
Specify \verb|-nfn| option to generate random number data without header.
EOF
scp=<<'EOF'
mnewrand -nfn l=5 S=1 o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)
